test_utils/utils/create_partition.py

```diff
@@ -0,0 +1,246 @@
+try:
+	import frappe
+	from frappe.utils import get_table_name
+except Exception as e:
+	raise (e)
+
+
+def add_custom_field(parent_doctype, partition_field):
+	parent_doctype_meta = frappe.get_meta(parent_doctype)
+	partition_docfield = parent_doctype_meta._fields.get(partition_field)
+
+	for child_doctype in [df.options for df in parent_doctype_meta.get_table_fields()]:
+
+		if frappe.get_all(
+			"Custom Field", filters={"dt": child_doctype, "fieldname": partition_field}
+		):
+			return
+
+		custom_field = frappe.get_doc(
+			{
+				"doctype": "Custom Field",
+				"dt": child_doctype,
+				"fieldname": partition_docfield.fieldname,
+				"fieldtype": partition_docfield.fieldtype,
+				"label": partition_docfield.label,
+				"options": partition_docfield.options,
+				"default": partition_docfield.default,
+				"read_only": 1,
+				"hidden": 1,
+			}
+		)
+		try:
+			custom_field.insert()
+		except Exception:
+			continue
+
+
+def populate_partition_fields(doc, event):
```

Comment:
@agritheory Maybe we'll need to run a patch or script in order to create the custom fields in child tables and populate them with data for the existing records.

Reply:
@fproldan If it's creating the field, we should run it at that time. I'd use the `bulk_insert` API.